### ISSUE TITLE
note left placed on the right side of sequence references

### ISSUE THREAD

**Issue by uwe83** [plantuml/plantuml#1682]
**Describe the bug**
Trying to place a note left of a sequence reference makes the note appear on the right consistently:

```
@startuml
opt test
    ref over A, B
        SomeSequence
    end ref
end

note left 
Appears on the right side, not on the left even though "note left" is used.
end note
@enduml
```

**To Reproduce**
https://www.plantuml.com/plantuml/uml/FOxH3OCm34J_FSM9RzGPe2im0Ie7H88uZHtcRuBQte_jfwUJ1_CvUpaFqUHmcWjgCbVenOonnrFmpQGd9xuBuuitP5oQAdMAH7NYuEgGCINEsQ0HlXCvRBl3mi8UrVhXs-N5TcdPTdJ_YWx1K8pBetMZSHdgLX_z0000

I tried in several contexts and this is consistently happening also in more complicated diagrams.

**Expected behavior**
Note should appear on left side

**Screenshots**
![image](https://github.com/plantuml/plantuml/assets/159607470/2861ba49-c718-4381-99c7-c9c9a254f13d)

**Desktop (please complete the following information):**
Seems to happen anywhere e.g. plantuml website see link above.

**Comment by jimnelson372** [comment]
Yes, in the original puma layout, this seems to be a problem with notes on groups.   With teoz turned on, the note doesn't appear at all, left or right.   

If the Opt/end is commented out, problems with Notes on Refs appear for puma: puma does put the note to the left of the Ref, but is drawing the ref borders with a larger width (as though it calculated the note to be inside the ref).  And for right side notes on the ref, puma doesn't draw the note at all, but again the Ref section is drawn very wide.  Teoz puts notes to the left and right of Ref blocks correctly.

 I have not yet looked at the code to see what is happening.

**Comment by yotambarnoy** [comment]
I've noticed a similar thing using the online server. I'm using a Component diagram, and all my 'left of' notes appear on the right. See [here](https://www.plantuml.com/plantuml/uml/XP7FRjH038VlynHMBi2XAUhUAeHQLk8FQcb58eHKLiZTUBD39vx8uooAKD-TdQZP7PX3Rv7z-PlOlukWA3fqpeG7opqATh3sNU-PM2iT7O7GMf5RHmdYB5CzzWHUTElDYsEdnwQntCBPwrZDbDSehUNODIzVGV6nhcz_S4Mo8oa00snL-n1RrSAOmnGKZV_DpcFpSsCTXR8V2_XZxZxKS6lN3uwWllowCdUNp5vHhMVuVdsrIe2LUJJ_22l5bk1iqdpYVb1uBzZHD7FuV7TrKorXRgUr8_IOQs4nQNy5pwkuL54BMZwjAUZfDtJmkRhvKiJ1QW6bs7QhvLjuRXbuKpvTv_-zuUGa7L_1EGoiA0btP9Q-wuL2e2R1rq_5_D5fR8dEtWigDLCn3imLapdJv-OZdOCGDoGcKqKkkV_yD_UMoQJ1pCeyaSZpZaIJv59sBYPW5YAK-m1YN7j6Q5ehVoY737ibSBHHy9iq0GEm74I8rOs08dP7uN1aQaf3t03xJ94VSxPvLhGSW8UkdvGu1h0Cn3jex3e6XZ8cuelzrz2vlm00)

**Comment by The-Lum** [comment]
Hi all,

_[This is an Issue Review]_ 👀 
This is now fixed.
 - Thanks @uwe83  for the report.
- Thanks PlantUML team for your work.

Here is a full example:
```puml
@startuml
opt test
    ref over A, B
        SomeSequence
    end ref
note left 
note left
end note
note right 
note right
end note
end

note left 
note left
end note
note right 
note right
end note
note left 
note left2
end note
@enduml
```
[![](https://img.plantuml.biz/plantuml/svg/fO_B3O0m34FlaR931k341C0AJ830V2HeeGJcfwK8zS0DdsphnLAANHkdnp8psLL5iIkJU3dqOayuAJEfOXLKsmKrjWEcHMnXkW37OAn2PlGgJAzd2ao8J-ccONo9EwI8Tqp9-O-fpuqy9Gfl_VSN)](https://editor.plantuml.com/uml/fO_B3O0m34FlaR931k341C0AJ830V2HeeGJcfwK8zS0DdsphnLAANHkdnp8psLL5iIkJU3dqOayuAJEfOXLKsmKrjWEcHMnXkW37OAn2PlGgJAzd2ao8J-ccONo9EwI8Tqp9-O-fpuqy9Gfl_VSN)

Regards.

**Comment by uwe83** [comment]
Cool, thank you so much for your Support!

The-Lum ***@***.***> schrieb am Sa., 11. Juli 2026, 10:21:
